pkg/action/executor/move_util.go

```diff
@@ -132,6 +250,219 @@ func movePod(client *kclient.Clientset, pod *api.Pod, nodeName string, retryNum
 	return xpod, nil
 }
 
+func deleteInvalidPendingPods(parent *unstructured.Unstructured, podClient v1.PodInterface,
+	podList *api.PodList) {
+	if parent == nil {
+		return
+	}
+	// Fetch the parents pods again
+	newPodList, err := parentsPods(parent, podClient)
+	if err != nil {
+		if err != nil {
+			glog.Errorf("Error getting pod list for %s: %s.", parent.GetKind(), parent.GetName())
+		}
+	}
+
+	// Find the pods which weren't there and if attributed to invalid
+	// turbo-scheduler, delete them.
+	for _, pod := range newPendingInvalidPods(newPodList, podList) {
+		err := commonutil.RetryDuring(defaultRetryLess, defaultRetryShortTimeout,
+			defaultRetrySleepInterval, func() error {
+				return podClient.Delete(pod.Name, &metav1.DeleteOptions{})
+			})
+		if err != nil {
+			glog.Errorf("Failed to delete pending pod: %s.", pod.Name)
+		}
+	}
+
+}
+
+func newPendingInvalidPods(list1, list2 *api.PodList) []*api.Pod {
```

Comment:
Nit: Maybe it's more readable to name the arguments more meaningful to something like `newPodList` and `existingPodList`

Reply:
Thanks for the suggestion, skipping this for now... :)